src/catalog/src/memory/objects.rs

```diff
@@ -1356,6 +1355,46 @@ impl CatalogItem {
         Ok(res)
     }
 
+    pub fn add_column(
+        &mut self,
+        name: ColumnName,
+        typ: ColumnType,
+        sql: RawDataType,
+    ) -> Result<(), PlanError> {
+        let CatalogItem::Table(table) = self else {
+            return Err(PlanError::Unsupported {
+                feature: "adding columns to a non-Table".to_string(),
+                discussion_no: None,
+            });
+        };
```

Comment:
I think this should probably be validated in planning (if it isn't already). I think these functions are usually called after the catalog commit, which means that we may have already committed bad data.